### ISSUE TITLE
Offline installation

### ISSUE THREAD

**Issue by mirkobrombin** [Vanilla-OS/iso-builder#301]
To achieve a full offline installation, we would have to populate a containers storage with desktop image. Since we have 3 different images, we have to generate 3 different ISOs:
- desktop
- nvidia
- vm

**Comment by taukakao** [comment]
Yes, but I have another suggestion:
We could probably make the user select a file in the regular installer containing the image. So a user could have two USBs or just load their image onto the flashed USB somewhere. 

This would be less complicated then always building three ISOs (plus a online ISO probably) and would open up the possibility of installing custom images right from the start.

**Comment by mirkobrombin** [comment]
Good idea, the user could have the tarball downloaded in addition to the ISO and have albius use it

**Comment by axtloss** [comment]
This can't replace isos with the images loaded already. if anything it should be an extra option since, we cannot expect users to have two USB drives

**Comment by mirkobrombin** [comment]
Yeah I was thinking of it as an extra option. I think the initial proposal is still the only solution to provide a full offline solution without asking the user to prepare 2 USB. The only problem is that generating 4 different ISOs (net, desktop, nvidia, vm) would require a good structure and automation/integration with the website.

In both cases providing a rootfs, whatever it is bundled with the ISO or not, is easier and would just require minor changes to Albius.

**Comment by taukakao** [comment]
Coudn't we do something similar to ventoy where the user could just load the tarball onto the flashed USB drive?

**Comment by axtloss** [comment]
That would require changes to the live iso which is just not worth it honestly, an extra option (like debian already has for external/offline apt repositories) is the one thats easiest to implement and most helpful

**Comment by albert-m-willemsen** [comment]
A true live cd would need an embedded container registry to be truely offline.

**Comment by albert-m-willemsen** [comment]
The problem with this is again manpower, you need someone to determine which packages and images are stable. And could also pose a security problem.

**Comment by taukakao** [comment]
Images can be extracted into an archive and imported again, that's not really an issue here.

**Comment by axtloss** [comment]
> A true live cd would need an embedded container registry to be truely offline.

It wouldn't, as tau already said, there are ways to import image archives.



> The problem with this is again manpower, you need someone to determine which packages and images are stable. And could also pose a security problem.

It doesn't, we don't use packages, and our selection of images is quite small, it's pretty clear what images are stable and should be included.
Also what security problem? It's not different than pulling the image from a registry

**Comment by CypherpunkSamurai** [comment]
Any updates? Installing vanilla os painfully slow.

**Comment by jdjxk** [comment]
Installation is too slow, waiting for offline installation.
